src/pheval_exomiser/post_process/assess_prioritisation.py

```diff
@@ -0,0 +1,471 @@
+
+                            (
+                                self.prioritisation_ranks.rank,
+                                self.prioritisation_ranks.variant,
+                            ) = rank, "_".join(
+                                [
+                                    variant.chrom,
+                                    str(variant.pos),
+                                    variant.ref,
+                                    variant.alt,
+                                ]
+                            )
+                            rank_stats.add_rank(rank)
+                            break
+                    break
+                self.prioritisation_ranks.rank = 0
+            self.prioritisation_ranks.record_variant_rank()
+
+
+@click.command()
+@click.option(
+    "--directory-list",
+    "-d",
+    required=True,
+    metavar="FILE",
+    help="A .txt file containing the full path of results directories,"
+    " with each new directory contained on a new line.",
+)
+@click.option(
+    "--phenopacket-dir",
+    "-p",
+    required=True,
+    metavar="PATH",
+    help="Full path to directory containing phenopackets.",
+)
+@click.option(
+    "--output-prefix",
+    "-o",
+    metavar="<str>",
+    required=True,
+    help=" Output file prefix. ",
+)
+@click.option(
+    "--ranking-method",
+    "-r",
+    metavar="<str>",
+    default="combinedScore",
+    show_default=True,
+    help="Ranking method for gene prioritisation. Options include: "
+    "combinedScore, phenotypeScore, variantScore and pValue",
+)
+@click.option(
+    "--threshold",
+    "-t",
+    metavar="<float>",
+    default=float(0.0),
+    required=False,
+    help="Score threshold.",
+)
+def assess_prioritisation(
+    directory_list, phenopacket_dir, ranking_method, output_prefix, threshold
+):
+    """Assesses percentage of top, top3 and top5 genes and variants found Exomiser results
+    from confirmed cases in phenopackets."""
+    gene_stats_writer = RankStatsWriter(output_prefix + "-gene_summary.tsv")
+    variants_stats_writer = RankStatsWriter(output_prefix + "-variant_summary.tsv")
+    directories = open(directory_list).read().splitlines()
+    gene_ranks, variant_ranks = None, None
+    for directory in directories:
+        prioritisation_ranks = PrioritisationRanks(directory=directory)
+        gene_rank_stats, variant_rank_stats = RankStats(), RankStats()
+        exomiser_json_results = files_with_suffix(Path(directory), ".json")
+        for exomiser_result in exomiser_json_results:
+            prioritisation_ranks.phenopacket = exomiser_result
+            phenopacket = os.path.join(
+                phenopacket_dir, exomiser_result.name.replace("-exomiser.json", ".json")
+            )
+            interpretations = PhenopacketReader(Path(phenopacket)).interpretations()
+            genes = PhenopacketReader.diagnosed_genes(interpretations)
+            variants = PhenopacketReader.diagnosed_variants(interpretations)
```

Comment:
If the `PhenopacketReader` class has a reference to the phenopacket and the interpretations, this would be a cleaner API:

```python
phenopacketReader = PhenopacketReader(Path(phenopacket))
genes = phenopacketReader.diagnosed_genes()
variants = phenopacketReader.diagnosed_variants()
```

or if you're keeping it as a stateless class:

```python
interpretations = PhenopacketReader.read_interpretations(Path(phenopacket))
genes = PhenopacketReader.read_diagnosed_genes(interpretations)
variants = PhenopacketReader.read_diagnosed_variants(interpretations)
```

Reply:
Ah yes, that does look better - I'll make that change